### PR TITLE
Implement the new Download Firefox CTA experiment

### LINKED ISSUE #10218: Experiment design for new Download Firefox CTA
We plan to conduct an experiment for the new Download Firefox CTA. Here are some assumptions/issues.

Assumptions:
- The experiment will compare the current UX with one single variation of the new UX.
- The experiment will be focused on the rate at which the Download Firefox CTA is clicked.
- The experiment will evaluate the final conversion rate (actual Firefox downloads) of the new CTA vs the existing CTA, but will not hypothesize on an increased rate of conversion. It is hypothesized, however, that there will be no regression in the conversion rate.
- A utm param will be used to allow mozilla.org to differentiate between the new CTA and the old CTA.
- A series of GA events will be used to record user behaviour wrt the new vs old CTAs, and will then be used for analysis.

Questions:
- What is our hypothesis for the increased click-rate for the new CTA?
- What utm parameter can/should we use to allow mozilla.org to differentiate between the new CTA and the old CTA? GA supports using the `utm_content` param to perform analysis, but that is the same param that is currently used by RTAMO, so there is a conflict there.
- Will the experiment be active for the mobile (`/android/`) site as well as the desktop (`/firefox/`) site?

Pinging @jvillalobos for validation and feedback.

### LINKED ISSUE #10252: Create callouts for the new Download Firefox button CTAs
For the experiment, the new CTAs will include callouts, which are elements which point to the existing buttons and draw the user's attention to them, from the Figma file at https://www.figma.com/file/7FnxRoHYpyEfMOIfob423s/Non-Firefox-Download?node-id=136%3A0:

Screen Shot 2021-03-22 at 11 07 59 AM

Screen Shot 2021-03-22 at 11 08 44 AM

### PR DESCRIPTION
**Pull request by bobsilverberg**
Fixes #10252 
Fixes #10218 

This does a few things:
- Enables the experiment for local dev and the dev server
- Creates two cohorts for the experiment, one will see the current UX, the other will see the new CTAs
- The new CTAs behave like so:
	- On the add-on detail page, a callout appears above the download button, and the button has new text
	- The current Download button is hidden from the header
	- On every page *other than* the add-on detail page, a dismissable banner appears encouraging the user to download Firefox
	- A GA event will be sent when a user dismisses the banner
- Different values for `utm_campaign` will be used depending on which cohort the user is in
- Different categories will be used for the GA event that is sent when a user click a Download Firefox button depending on which cohort the user is in

### CODE REVIEW

**Comment by bobsilverberg** [comment]
@willdurand I still need to write/update tests, but I have added a commit that adds a wrapper as you requested.

**Comment by bobsilverberg** [comment]
@willdurand After doing that I realized we don't need the wrapper at all, we can just use `InstallButtonWrapper` as the wrapper. Or do you also need access to `InstallButtonWrapper` without the experiment code?

**Comment by willdurand** [comment]
> @willdurand After doing that I realized we don't need the wrapper at all, we can just use `InstallButtonWrapper` as the wrapper. Or do you also need access to `InstallButtonWrapper` without the experiment code?

I don't think I need to use the `InstallButtonWrapper` but don't we need the download button in the header for non-detail pages?

**Comment by bobsilverberg** [comment]
> > @willdurand After doing that I realized we don't need the wrapper at all, we can just use `InstallButtonWrapper` as the wrapper. Or do you also need access to `InstallButtonWrapper` without the experiment code?
> 
> I don't think I need access to the `InstallButtonWrapper` but don't we need the download button in the header for non-detail pages?

No, a decision was made that for users in the "new version" cohort, the button will be hidden in the header, so we don't need an experimental version of the button for the header, we can just use the default one.

Having created this wrapper, I kind of like it. It's a good place to encapsulate the constants for the variants and the experiment config, so I might just keep it and use it to wrap `InstallButtonWrapper` instead of wrapping `GetFirefoxButton` (because `InstallButtonWrapper` also needs access to the experiment).

**Comment by bobsilverberg** [comment]
I requested review, but then realized I need to make a few more changes.

**Comment by bobsilverberg** [comment]
This is ready again for a review. Looks like @eviljeff won the lottery this time. This is a pretty huge patch, but I would ideally like to land it on dev next week. I am available to discuss anything in the patch that might be hard to follow.

File: src/amo/experiments/downloadCtaExperiment20210404.js [hunk 1]
```diff
@@ -0,0 +1,11 @@
+/* @flow */
```
Comment (willdurand) [inline src/amo/experiments/downloadCtaExperiment20210404.js:1]: I think having the same `id` and filename would make sense here.

File: src/amo/components/GetFirefoxButton/index.js [hunk 2]
```diff
@@ -37,45 +47,64 @@ export type Props = {|
   addon?: AddonType,
   buttonType: GetFirefoxButtonTypeType,
   className?: string,
+  useNewVersion?: boolean,
```
Comment (willdurand) [inline src/amo/components/GetFirefoxButton/index.js:50]: The current state is probably okay but some of the mocks have an icon in the yellow thing: https://www.figma.com/file/7FnxRoHYpyEfMOIfob423s/Non-Firefox-Download?node-id=323%3A61

File: src/amo/components/GetFirefoxButton/styles.scss [hunk 3]
```diff
@@ -10,3 +13,47 @@
...
+
+// New styles for the CTA experiment
+.GetFirefoxButton--new {
+  .GetFirefoxButton-button {
+    margin: auto;
+    white-space: nowrap;
+    width: 100%;
+
+    @include respond-to(medium) {
+      align-self: flex-end;
```
Comment (willdurand) [inline src/amo/components/GetFirefoxButton/styles.scss:25]: Is it working okay in RTL mode?

File: src/amo/components/GetFirefoxButton/index.js [hunk 4]
```diff
@@ -37,45 +47,64 @@ export type Props = {|
...
+|};
+
+export type DeafultProps = {|
```
Comment (eviljeff) [inline src/amo/components/GetFirefoxButton/index.js:53]: "DefaultProps"?

File: src/amo/components/GetFirefoxButton/index.js [hunk 5]
```diff
@@ -84,14 +113,41 @@ export const GetFirefoxButtonBase = (
...
+        buttonText = supportsRTAMO
+          ? downloadTextForRTAMO
+          : i18n.gettext('Download Firefox');
+        calloutText =
+          addon.type === ADDON_TYPE_STATIC_THEME
+            ? i18n.gettext(`You'll need Firefox to use this theme`)
+            : i18n.gettext(`You'll need Firefox to use this extension`);
+      }
       puffy = true;
+      utmCampaign = `amo-fx-cta-${addon.id}-${variant}`;
```
Comment (eviljeff) [inline src/amo/components/GetFirefoxButton/index.js:142]: mmm, tbh I'd prefer that these `utmCampaign` values were defined in a single place, or built in a utility function.  But for an experiment it's fine...?

File: src/amo/components/GetFirefoxButton/index.js [hunk 6]
```diff
@@ -84,14 +113,41 @@ export const GetFirefoxButtonBase = (
...
+
       utmContent = addon.guid ? `rta:${_encode(addon.guid)}` : '';
       break;
     }
     case GET_FIREFOX_BUTTON_TYPE_HEADER: {
       buttonText = i18n.gettext('Download Firefox');
       micro = true;
+      utmCampaign = `amo-fx-cta-${variant}`;
```
Comment (eviljeff) [inline src/amo/components/GetFirefoxButton/index.js:150]: same comment about `utmCampaign` values here

**Review by eviljeff — APPROVED** [review]

File: src/amo/components/GetFirefoxButton/index.js (hunk 2 above)
Comment (bobsilverberg) [reply src/amo/components/GetFirefoxButton/index.js:50]: > The current state is probably okay but some of the mocks have an icon in the yellow thing: [figma.com/file/7FnxRoHYpyEfMOIfob423s/Non-Firefox-Download?node-id=323%3A61](https://www.figma.com/file/7FnxRoHYpyEfMOIfob423s/Non-Firefox-Download?node-id=323%3A61)

Those are quite old. The newer ones don't have the icon, and screenshots of the version from this PR have been approved at https://github.com/mozilla/addons-frontend/issues/10252#issuecomment-807440995

File: src/amo/components/GetFirefoxButton/styles.scss (hunk 3 above)
Comment (bobsilverberg) [reply src/amo/components/GetFirefoxButton/styles.scss:25]: It is! I guess it makes sense that `flex-end` would be language-direction-aware, as it's flex-*end*, not flex-*right*.

File: src/amo/components/GetFirefoxButton/index.js (hunk 4 above)
Comment (bobsilverberg) [reply src/amo/components/GetFirefoxButton/index.js:53]: 🤦

File: src/amo/components/GetFirefoxButton/index.js (hunk 5 above)
Comment (bobsilverberg) [reply src/amo/components/GetFirefoxButton/index.js:142]: This is a good idea. I'll do this.

File: src/amo/components/GetFirefoxButton/index.js (hunk 2 above)
Comment (willdurand) [reply src/amo/components/GetFirefoxButton/index.js:51]: @bobsilverberg is it expected to have a different yellow?

**Review by willdurand — CHANGES_REQUESTED** [review]
This implements outdated mocks so we should update the UI a bit.

File: src/amo/components/GetFirefoxButton/index.js (hunk 2 above)
Comment (bobsilverberg) [reply src/amo/components/GetFirefoxButton/index.js:51]: Based on the discussion in Slack I have updated it to use the new yellow and the icon.

**Comment by bobsilverberg** [comment]
@willdurand New screenshots can be seen at https://github.com/mozilla/addons-frontend/issues/10252#issuecomment-816793871.

**Comment by bobsilverberg** [comment]
@willdurand This has an r+ from @eviljeff. Were you going to take another look as well?

**Comment by willdurand** [comment]
> @willdurand This has an r+ from @eviljeff. Were you going to take another look as well?

Yes

**Comment by bobsilverberg** [comment]
> > @willdurand This has an r+ from @eviljeff. Were you going to take another look as well?
> 
> Yes

Ok, thanks. We are hoping to land it on dev this week.

**Review by willdurand — APPROVED** [review]
